test/index.test.ts

```diff
@@ -183,15 +236,450 @@ describe('HipThrusTS', () => {
+          typeof pipedAtoBC.attachData
+        >
+          ? true
+          : false;
+        type ReturnAssignableFromCorrect = ReturnType<
+          typeof pipedAtoBC.attachData
+        > extends CorrectReturnValue
+          ? true
+          : false;
+        // @ts-expect-error
+        const paramAssignableToCorrectShouldFail: ParamAssignableToCorrect = false;
+        // @ts-expect-error
+        const paramAssignableFromCorrectShouldFail: ParamAssignableFromCorrect = false;
+        // @ts-expect-error
+        const returnAssignableToCorrectShouldFail: ReturnAssignableToCorrect = false;
+        // @ts-expect-error
+        const returnAssignableFromCorrectShouldFail: ReturnAssignableFromCorrect = false;
+      });
+      it('attaches properly typed data from left sync data attacher and right not fully covered sync data attacher', () => {
+        const left = {
+          attachData(context: { a: string }) {
+            return { b: 4 };
+          },
+        };
+
+        const rightPartiallyCovered = {
+          attachData(context: { b: number; other: string }) {
+            return { c: 4 };
+          },
+        };
+
+        const pipedAOtoBC1 = HTPipeAttachData(left, rightPartiallyCovered);
+
+        interface CorrectParam {
+          a: string;
+          other: string;
+        }
+        type CorrectReturnValue = PromiseOrSync<{ b: number; c: number }>;
+        type ParamAssignableToCorrect = CorrectParam extends Parameters<
+          typeof pipedAOtoBC1.attachData
+        >[0]
+          ? true
+          : false;
+        type ParamAssignableFromCorrect = Parameters<
+          typeof pipedAOtoBC1.attachData
+        >[0] extends CorrectParam
+          ? true
+          : false;
+        type ReturnAssignableToCorrect = CorrectReturnValue extends ReturnType<
+          typeof pipedAOtoBC1.attachData
+        >
+          ? true
+          : false;
+        type ReturnAssignableFromCorrect = ReturnType<
+          typeof pipedAOtoBC1.attachData
+        > extends CorrectReturnValue
+          ? true
+          : false;
+        // @ts-expect-error
+        const paramAssignableToCorrectShouldFail: ParamAssignableToCorrect = false;
+        // @ts-expect-error
+        const paramAssignableFromCorrectShouldFail: ParamAssignableFromCorrect = false;
+        // @ts-expect-error
+        const returnAssignableToCorrectShouldFail: ReturnAssignableToCorrect = false;
+        // @ts-expect-error
+        const returnAssignableFromCorrectShouldFail: ReturnAssignableFromCorrect = false;
+      });
+      it('no attaches data from left sync data attached and right not covered sync data attacher', () => {
+        const left = {
+          attachData(context: { a: string }) {
+            return { b: 4 };
+          },
+        };
+        const rightNotCovered = {
+          attachData(context: { other: string }) {
+            return { c: 4 };
+          },
+        };
+
```

Comment:
why should this be an error?  If I'm not mistaken, we should end up with something that takes {a: string, other: string} and returns {b: 4, c: 4}.
- [ ] something seems wrong here.

Reply:
the problem was with out typing
i decide to avoid this problem by checking if TRight attach data params have at least one key from TLeft outputs before putting that type to TLeft HasAttachData
just in case error case with bad left working like supposed
also if we have at least one same key in left outputs and right inputs it returns Partial<Parameters<TRight['attachData']>[0]>